packages/frontend/src/features/metricsCatalog/components/visualization/MetricsVisualization.tsx

```diff
@@ -140,12 +211,24 @@ const MetricsVisualization: FC<Props> = ({
     const legendConfig = useMemo(() => {
         if (comparison.type === MetricExplorerComparison.NONE) return null;
         if (comparison.type === MetricExplorerComparison.DIFFERENT_METRIC) {
-            return [results?.metric.label, results?.compareMetric?.label];
+            return {
+                metric: { name: 'metric', label: results?.metric.label },
+                compareMetric: {
+                    name: 'compareMetric',
+                    label: results?.compareMetric?.label,
+                },
+            };
         }
         if (comparison.type === MetricExplorerComparison.PREVIOUS_PERIOD) {
-            return [results?.metric.label, 'Previous period'];
+            return {
+                metric: { name: 'metric', label: results?.metric.label },
+                compareMetric: {
+                    name: 'compareMetric',
+                    label: 'Previous period',
```

Comment:
The response from running the metric query already has a label for `PREVIOUS_PERIOD` compare metric which matches `Previous period` in (packages/common/src/utils/metricsExplorer.ts) so this can be `results?.compareMetric?.label` and we can make this a `switch` statement that returns the same object for both comparison types

![image](https://github.com/user-attachments/assets/d62d3cbb-3bad-4758-9afe-3ad7e891661e)

Reply:
ahh yes thank you, v smart

Reply:
https://github.com/lightdash/lightdash/pull/12816/commits/fba9f38718d33f12ef7ae5bda6a4e8fafe19cddc

done here and tested